source/io/module_eggShell.js

```diff
@@ -373,6 +373,14 @@
         };
 
         Module.eggShell.executeSlices = publicAPI.eggShell.executeSlices = function (slices) {
+            // compatibility non-async entry point for callers which do not expect negative return values
+            var result = EggShell_ExecuteSlices(v_userShell, slices);
+            return (result !== 0) ? 1 : 0;
+        };
+
+        Module.eggShell.executeSlicesAsync = publicAPI.eggShell.executeSlicesAsync = function (slices) {
```

Comment:
I don't think we need to maintain both functions at this time. Our only client is desktop and I think it is more important at this time to build up a clean api.

Users of the npm js version are very likely to be using package management and I think it is sufficient to create a major version bump to signify the API change and include the change in the description.

If there is consensus with @cglzaguilar and @sanmut to preserve both endpoints then I think we need a different name. Usually the Async version of a JS function is expected to either take a callback or return a promise that resolves asynchronously to the current stack. This function runs synchronously and returns a value synchronously that is consumed on the same stack. A better name might be "executeSlicesAsNecessary" or "executeSlicesUntilWait". We should also add a deprecated comment to the executeSlices function and add a run once comment log that states the function is deprecated (use a flag to print a warning on first call).

Reply:
I don't think we need to keep both endpoints if there is not need.

Reply:
I'm fine with executeSlicesUntilWait or similar.  But I was envisioning this change to be *very* temporary;  @cglzaguilar had issues running this on the CI using the modified executeSlices which can return negative values, and I wanted to see if the issues were solely due to the negative return values or not.   If so, it would be due to code which tested if the return value > 0 rather than !== 0.  There's no reason for such callers to have made this assumption, so I wouldn't be surprised if they were only written this way to avoid triggering that terrible eslint rule.

Assuming this code passes the CI, then rather than submitting this and introducing the fork, if you can go ahead and fix all executeSlices in the TFS code to test for non-zero instead of > 0  first (which is still compatible with the previous incarnation of executeSlices and more correct), I can just leave the single entry point.

Reply:
The TFS code has one usage of executeSlices to update in `niEggShell.js`:

```
retVal = this._vireo.eggShell.executeSlices(1000);
...
executionState.finished = (retVal > 0 ? false : true);
```

which can be updated as part of the integration of the new version of Vireo into TFS. So yes please remove the additional entry point that was added from this file and all the other JS files that were updated to use it.

In summary:
1. We will have just the `executeSlices` function with the new behavior
2. When the package is updated in TFS, that changeset will include the change for how the return value of executeSlices is used

Reply:
We just talked about it but I'm also okay with having the function name updated to executeSlicesUntilWait. The new name describes the behavior more precisely and having the name change is a more noticeable breaking change for users updating the package which is a win-win to me.